code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java

```diff
@@ -470,4 +473,20 @@ public void doWithLockNoResult() {
             }
         });
     }
+
+    @Override
+    public void setDnsToken(Service service) {
+        String dnsToken = generateNewDnsToken();
+        DataAccessor.fields(service).withKey(ServiceDiscoveryConstants.FIELD_DNSTOKEN).set(dnsToken);
+        objectManager.persist(service);
+    }
+
+    private String generateNewDnsToken() {
+        SecureRandom random = new SecureRandom();
```

Comment:
Just use `io.cattle.platform.iaas.api.filter.apikey.ApiKeyFilter.generateKeys()[1]` instead.  That is the same logic we use for API keys.